File: Rust/Hamming Distance/hamming-distance.rs
fn hamming_distance(str1: &str, str2: &str) -> Option<usize> {
    if str1.len() != str2.len() {
        return None;
    }
    Some(
        str1.chars()
            .zip(str2.chars())
            .fold(0, |acc, (c1, c2)| if c1 != c2 { acc + 1 } else { acc }),
    )
}

fn main() {
    let mut str1 = String::new();
    let mut str2 = String::new();
    println!("Enter a first string:");
    std::io::stdin().read_line(&mut str1);
    println!("Enter a second string:");
    std::io::stdin().read_line(&mut str2);
    match hamming_distance(&str1, &str2) {
        None => println!("String length mismatch: can't compute Hamming Distance"),
        Some(distance) => println!(
            "The hamming distance between these two strings is {}",
            distance
        ),
    }
}

File: algorithms/C++/HammingDistance/HammingDistance.cpp
#include <iostream>
#include <string>

using namespace std;

int hammingDistance(const string str1, const string str2)
{
	//IF LENGTH BEETWEEN STRINGS IS DIFFERENT
	if(str1.length() != str2.length())
		return -1;
	else
	{
		int dist = 0;
		
		//COMPARE PARALLEL CHARACTERS
		for(unsigned i = 0; i < str1.length(); i++)
			//IF DIFFERENT, INCREMENT DISTANCE VARIABLE
			if(str1[i] != str2[i])
				++dist;
		
		return dist;
	}
		
}

int main()
{
	string str1, str2;
	
	//INPUT TWO STRINGS
	cout<<"Insert first string: ";
	cin>>str1;
	
	cout<<"Insert second string (of equal length): ";
	cin>>str2;
	
	//OUTPUT HAMMING DISTANCE
	cout<<"Hamming distance between these two strings (-1 if length is different): "<<hammingDistance(str1, str2)<<endl;
	
	return 0;
}
